File: PHP/QuadradoAoCuboTest.php
<?php
/**
 * @version PHP
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ ./vendor/bin/phpunit NomeDaClasseTest.php
 */
namespace PHP;

use PHPUnit\Framework\TestCase;
require ("QuadradoAoCubo.php");

class QuadradoAoCuboTest extends TestCase {

    public function testQuadradoAoCubo() {
        $obj = new QuadradoAoCubo();
        //Test 1
        $want = [[1,1,1], [2,4,8], [3,9,27], [4,16,64], [5,25,125]];
        $got = $obj->loadSquareAndCumbe( 5 );
        $this->assertEquals($want, $got);
    }

}

File: GO/QuadradoAoCubo_test.go
/**
 * @version GO
 * @author MARIO SAM <eu@mariosam.com.br>
 * @see I would love to work with you instead solving web code tests: hire me!
 * $ go test
 */
package GO

import (
	"reflect"
	"testing"
)

func TestQuadradoAoCubo(t *testing.T) {
	tables := []struct {
		N    int
		want [][]int
	}{
		{5, [][]int{{1, 1, 1}, {2, 4, 8}, {3, 9, 27}, {4, 16, 64}, {5, 25, 125}}},
	}

	for _, table := range tables {
		got := loadSquareAndCumbe(table.N)

		if !reflect.DeepEqual(got, table.want) {
			t.Errorf("Waiting for this %v but the return was this: %v", table.want, got)
		}
	}
}
